modes/play/objects/obj.js

```diff
@@ -9,13 +9,17 @@ class Obj {
     this.mode = mode;
     this.collisionRecord = {};
     this.cooldowns = {};
+    this.status = {
+      action: 'default',
+      animationFrame: 0,
+    };
   }
 
   get sprite() {
-    if (!this.spritePath) { return null }
-    if (this.image) { return this.image }
-    this.image = new Image()
-    this.image.src = this.spritePath;
+    const currentSprite = this.spriteBank[this.status.action][this.status.animationFrame];
+    if (!currentSprite) { return null }
```

Comment:
What about `if (!currentSprite) { return this.spriteBank[this.status.action][0] }`? That might solve the "Jump animation breaks if the animation frame isn't 0" problem.